clients/drcachesim/tracer/output.cpp

```diff
@@ -767,7 +781,10 @@ output_buffer(void *drcontext, per_thread_t *data, byte *buf_base, byte *buf_ptr
             DR_ASSERT(
                 is_ok_to_split_before(instru->get_entry_type(pipe_start + header_size),
                                       instru->get_entry_size(pipe_start + header_size)));
-            pipe_start = atomic_pipe_write(drcontext, pipe_start, pipe_end,
+            // If the following assert triggers, we found too many entries without an
+            // ok-to-split point.
```

Comment:
Same comment: move this to the assert as a string so it is instantly visible to the user/dev and isn't just in the code.

Reply:
Done